src/backend/TrafficCourts/Citizen.Service/Services/Tickets/Search/Rsi/RoadSafetyTicketSearchService.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Collections.Concurrent;
+using TrafficCourts.Citizen.Service.Logging;
+using TrafficCourts.Citizen.Service.Services.Tickets.Search.Common;
+
+namespace TrafficCourts.Citizen.Service.Services.Tickets.Search.Rsi
+{
+    public class RoadSafetyTicketSearchService : ITicketInvoiceSearchService
+    {
+        private readonly IRoadSafetyTicketSearchApi _api;
+        private readonly ILogger<RoadSafetyTicketSearchService> _logger;
+
+        public RoadSafetyTicketSearchService(IRoadSafetyTicketSearchApi api, ILogger<RoadSafetyTicketSearchService> logger)
+        {
+            _api = api ?? throw new ArgumentNullException(nameof(api));
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+
+        public async Task<IEnumerable<Invoice>> SearchAsync(string ticketNumber, TimeOnly issuedTime, CancellationToken cancellationToken)
+        {
+            using var activity = Diagnostics.Source.StartActivity("RSI Ticket Search");
+
+            RawTicketSearchResponse? response = null;
+
+            try
+            {
+                var parameters = new GetTicketParams 
+                { 
+                    TicketNumber = ticketNumber, 
+                    Prn = "10006", 
+                    IssuedTime = $"{issuedTime.Hour:d2}{issuedTime.Minute:d2}"
+                };
+
+                response = await _api.GetTicket(parameters, cancellationToken);
+            }
+            catch (HttpRequestException ex)
+            {
+                activity?.SetStatus(System.Diagnostics.ActivityStatusCode.Error);
+                var innerException = ex.InnerException;
```

Comment:
unfinished code. was looking at conditions when the request could not be made ie not on VPN which causes a socket connection error. Should elaborate on the reasons we may get exceptions here and add retry where applicable.